test/suite/hook_nonstandalone.py

```diff
@@ -51,30 +51,18 @@ def get_platform_api(self):
         return self.platform_api
 
     # Is this test one we should skip?
-    def skip_test(self, test):
-        # Skip any test that contains one of these strings as a substring
-        skip = [
-                # Skip all tests that do timestamped truncate operations.
-                "test_checkpoint25.test_checkpoint",
-                "test_rollback_to_stable34.test_rollback_to_stable",
-                "test_rollback_to_stable36.test_rollback_to_stable36",
-                "test_truncate09.test_truncate09",
-                "test_truncate15.test_truncate15",
+    def skip_reason(self, test):
+        # There are no general categories of tests to skip. 
+        # Individual tests are skipped via the skip_for_hook decorator
+        return None
 
-                # This group fail within Python for various, sometimes unknown, reasons.
-                "test_bug018.test_bug018"
-                ]
-
-        for item in skip:
-            if item in str(test):
-                return True
-        return False
-
-    # Remove tests that won't work on non-standalone build
-    def filter_tests(self, tests):
-        new_tests = unittest.TestSuite()
-        new_tests.addTests([t for t in tests if not self.skip_test(t)])
-        return new_tests
+    # Skip tests that won't work on non-standalone build
+    def register_skipped_test(self, tests):
+        for t in tests:
+            skip_reason = self.skip_reason(t)
+            if skip_reason is not None:
+                wttest.register_skipped_test(t, "nonstandalone", skip_reason)
+        return tests
```

Comment:
Is the provided `tests` arg directly modified by this function, that's why we can return it?

Reply:
Yep, changed as per Don's comment in [17b39ee](https://github.com/wiredtiger/wiredtiger/pull/9764/commits/17b39ee2b77ba4b8b4ac33c1721796644b0c6342)